### PR TITLE
Use libarchive for extracting archives

### PR DESCRIPTION
**Pull request by mschwan-phytec**
Use libarchive for extracting archives instead of calling tar in a separate process.

Prefix all source files with "pu-" to avoid name clashes with other libraries, particularly when including headers.

### CODE REVIEW

File: src/pu-archive.c [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    gint flags;
+    gint read;
+    gint write;
+
+    flags = ARCHIVE_EXTRACT_TIME |
+            ARCHIVE_EXTRACT_PERM |
+            ARCHIVE_EXTRACT_OWNER |
+            ARCHIVE_EXTRACT_ACL |
+            ARCHIVE_EXTRACT_FFLAGS;
+
```
Comment (landerweit-phytec) [inline src/pu-archive.c:58]: The parameter checks are missing here:
```
g_return_val_if_fail(filename != NULL, FALSE);
g_return_val_if_fail(dest != NULL, FALSE);
g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
```

File: src/pu-archive.c [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    read = archive_read_open_filename(ar, filename, 1024);
+
+    if (read != ARCHIVE_OK) {
+        g_set_error(error, PU_ERROR, PU_ERROR_ARCHIVE_READ,
+                    "Failed reading archive '%s'", filename);
+        return FALSE;
+    }
+
+    old_dest = g_get_current_dir();
+    g_chdir(dest);
```
Comment (landerweit-phytec) [inline src/pu-archive.c:74]: Check the return value in case `dest` does not exist.

**Review by landerweit-phytec — CHANGES_REQUESTED** [review]

File: src/pu-archive.c (hunk 2 above)
Comment (mschwan-phytec) [reply src/pu-archive.c]: Done.

File: src/pu-archive.c (hunk 1 above)
Comment (mschwan-phytec) [reply src/pu-archive.c:62]: Fixed.

**Comment by mschwan-phytec** [comment]
I'm setting this PR to draft mode for now. While thinking about our provisioning image and its limited space with initramfs, I think we should reconsider adding libarchive for this simple task (extracting archives). Using `tar` has the benefit that it is available already on all machines. With libarchive we may require about 1MB of additional space (looking at the .so on my Arch system).

**Comment by mschwan-phytec** [comment]
The renaming of source files may have to be separated, of course.

**Comment by mschwan-phytec** [comment]
Closing for now, as we don't want to introduce another dependency on libarchive at this point.

**Pull request closed without merging**